termux-app/terminal-term/src/main/java/com/termux/app/TermuxActivity.java

```diff
@@ -313,6 +323,49 @@ public void onPageSelected(int position) {
         mBellSoundId = mBellSoundPool.load(this, R.raw.bell, 1);
     }
 
+    /** Parse intent for connection parameters **/
+    private String username = "";
+    private String hostname = "";
+    private String port = "";
+    private String sessionName = "";
+
+    private void parseUserlandIntent(String intentData) {
+        String regexPattern = "ssh://([\\w\\W]+)@([\\w\\W]+):([\\d]+)/#([\\w\\W]+)";
+        Pattern pattern = Pattern.compile(regexPattern);
+
+        try {
+            Matcher matcher = pattern.matcher(intentData);
+            if (matcher.groupCount() < 4) {
+                return;
+            }
+
+            if (matcher.find()) {
+                username = matcher.group(1);
+                hostname = matcher.group(2);
+                port = matcher.group(3);
+                sessionName = matcher.group(4);
+            }
+        } catch (Exception e) {
+            showErrorAndGoBackToUserland("Error occurred while parsing intent data with regex: " + e);
+        }
+    }
+
+    public void showErrorAndGoBackToUserland(String errorMessage) {
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
+
+        alertDialogBuilder.setTitle("An error has occurred")
+                .setMessage(errorMessage)
+                .setCancelable(true)
+                .setPositiveButton("Exit",new DialogInterface.OnClickListener() {
```

Comment:
Same with the "exit".

Reply:
This or something similar should already exist in the UserLAnd resources.

Reply:
Same as above